contracts/Linea_SpokePool.sol

```diff
@@ -139,27 +133,32 @@ contract Linea_SpokePool is SpokePool {
     function _bridgeTokensToHubPool(uint256 amountToReturn, address l2TokenAddress) internal override {
         // Linea's L2 Canonical Message Service, requires a minimum fee to be set.
         uint256 minFee = minimumFeeInWei();
-        // We require that the caller pass in the fees as msg.value instead of pulling ETH out of this contract's balance.
-        // Using the contract's balance would require a separate accounting system to keep LP funds separated from system funds
-        // used to pay for L2->L1 messages.
-        require(msg.value == minFee, "MESSAGE_FEE_MISMATCH");
 
         // SpokePool is expected to receive ETH from the L1 HubPool, then we need to first unwrap it to ETH and then
         // send ETH directly via the Canonical Message Service.
         if (l2TokenAddress == address(wrappedNativeToken)) {
+            // We require that the caller pass in the fees as msg.value instead of pulling ETH out of this contract's balance.
+            // Using the contract's balance would require a separate accounting system to keep LP funds separated from system funds
+            // used to pay for L2->L1 messages.
+            require(msg.value == minFee, "MESSAGE_FEE_MISMATCH");
+
             // msg.value is added here because the entire native balance (including msg.value) is auto-wrapped
             // before the execution of any wrapped token refund leaf. So it must be unwrapped before being sent as a
             // fee to the l2MessageService.
             WETH9Interface(l2TokenAddress).withdraw(amountToReturn + msg.value); // Unwrap into ETH.
             l2MessageService.sendMessage{ value: amountToReturn + msg.value }(withdrawalRecipient, msg.value, "");
         }
         // If the l1Token is USDC, then we need sent it via the USDC Bridge.
-        else if (l2TokenAddress == l2UsdcBridge.usdc()) {
-            IERC20(l2TokenAddress).safeIncreaseAllowance(address(l2UsdcBridge), amountToReturn);
-            l2UsdcBridge.depositTo{ value: msg.value }(amountToReturn, withdrawalRecipient);
+        else if (l2TokenAddress == address(usdcToken) && _isCCTPEnabled()) {
```

Comment:
Although I'm sure there are plenty of other gas optimizations we can do, wydt about removing the `_isCCTPEnabled` check here so we don't do the extra access to `cctpTokenMessenger`? Linea either has native USDC or USDC.e, but will never have both concurrently, so we may be able to just get rid of this check.

Reply:
I can see this being bad if there are other "Linea-like" L2s that we need to support.

Reply:
> I can see this being bad if there are other "Linea-like" L2s that we need to support.

This is one consideration yeah, i'd prefer the extra safety for now. The adapters don't need to be super gas efficient as they're not called that regularly.